docs/pages/api-reference/liveblocks-react.mdx

```diff
@@ -10,21 +10,1134 @@ alwaysShowAllNavigationLevels: false
+presence, and this is readable for all other connected users. Set your initial
+Presence value by using `initialPresence`.
+
+```tsx highlight="7-12"
+import { RoomProvider } from "@liveblocks/react/suspense";
+
+function App() {
+  return (
+    <RoomProvider
+      id="my-room"
+      initialPresence={{
+        cursor: null,
+        colors: ["red", "purple"],
+        selection: {
+          id: 72426,
+        }}
+    >
+      {/* children */}
+    </RoomProvider>
+  );
+}
+```
+
+Each user’s Presence resets every time they disconnect, as this is only meant
+for temporary data. Any JSON-serializable object is allowed (the `JsonObject`
+type).
+
+#### Setting initial Storage [#setting-initial-storage]
+
+Storage is used to store permanent data that’s used in your application, such as
+shapes on a whiteboard, nodes on a flowchart, or text in a form. The first time
+a room is entered, you can set an initial value by using `initialStorage`. Note
+that this value is only read a single time.
+
+```tsx highlight="9-15"
+import { LiveList, LiveObject } from "@liveblocks/client";
+import { RoomProvider } from "@liveblocks/react/suspense";
+
+function App() {
+  return (
+    <RoomProvider
+      id="my-room"
+      initialPresence={}
+      initialStorage={{
+        title: "Untitled",
+        names: new LiveList(["Steven", "Guillaume"])
+        shapes: new LiveMap([
+          ["g9shu0", new LiveObject({ type: "rectangle", color: "red" })],
+          ["djs3g5", new LiveObject({ type: "circle", color: "yellow" })],
+        ]),
+      }}
+    >
+      {/* children */}
+    </RoomProvider>
+  );
+}
+```
+
+Any
+[conflict-free live structures](/docs/api-reference/liveblocks-client#Storage)
+and JSON-serializable objects are allowed (the `LsonObject` type).
+
 ### createRoomContext
 
-Creates a [`RoomProvider`][] and a set of typed hooks. We recommend using it in
-`liveblocks.config.ts` and re-exporting your typed hooks like below.
+<Banner>
 
-```tsx file="liveblocks.config.ts"
+This used to be the default way to start your app, but now it’s recommend for
+advanced usage only. We generally recommend using ... TODO ... unless you need
+to define multiple room types in your application.
+
+</Banner>
+
+Creates a [`RoomProvider`][] and a set of typed hooks to use in your app. We
+generally recommend typing your app using the newer method instead. When using
+`createRoomContext` it can be helpful to use it in `liveblocks.config.ts` and
+re-export your typed hooks as below.
+
+```tsx file="liveblocks.config.ts" highlight="15-23"
+import { createClient } from "@liveblocks/client";
+import { createRoomContext } from "@liveblocks/react";
+
+const client = createClient({
+  // publicApiKey: "",
+  // authEndpoint: "/api/liveblocks-auth",
+});
+
+type Presence = {};
+type Storage = {};
+type UserMeta = {};
+type RoomEvent = {};
+type ThreadMetadata = {};
+
+export const {
+  RoomProvider,
```

Comment:
Not sure where we should mention it, but the `RoomProvider` exported here does not need to be wrapped in a `LiveblocksProvider`, like it normally would have to. This is true only for the factory-exported one, since it's already bound to the passed-in `client` instance.